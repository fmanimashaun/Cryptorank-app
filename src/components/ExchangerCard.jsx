import React from 'react';
import PropTypes from 'prop-types';
import { Card, Button } from 'react-bootstrap';
import Style from 'assets/scss/exchangercard.module.scss';

const ExchangerCard = ({ exchanger }) => (
  <Card className={Style.card}>
    <Card.Img
      src={exchanger.image}
      className={Style.image}
      alt={exchanger.name}
    />
    <Card.Body className={Style.body}>
      <Card.Title>{exchanger.name}</Card.Title>
      <Card.Text className={Style.text}>{`Ranking: ${exchanger.trust_score_rank}`}</Card.Text>
      <Button variant="primary" className={Style.btn}>
        See Details
      </Button>
    </Card.Body>
  </Card>
);

ExchangerCard.propTypes = {
  exchanger: PropTypes.shape({
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    trust_score_rank: PropTypes.number.isRequired,
  }).isRequired,
};

export default ExchangerCard;
