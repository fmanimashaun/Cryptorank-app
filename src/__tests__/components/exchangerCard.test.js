import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ExchangerCard from 'components/exchangerCard';

describe('ExchangerCard Component', () => {
  const mockExchanger = {
    image: 'test-image-url',
    name: 'Test Exchanger',
    trust_score_rank: 1,
    id: 'test-id',
  };

  it('renders without crashing', () => {
    const { getByText } = render(
      <ExchangerCard exchanger={mockExchanger} showDetails={() => {}} />,
    );
    expect(getByText('Test Exchanger')).toBeInTheDocument();
    expect(getByText('1')).toBeInTheDocument();
  });

  it('displays the correct image', () => {
    const { getByAltText } = render(
      <ExchangerCard exchanger={mockExchanger} showDetails={() => {}} />,
    );
    expect(getByAltText('Test Exchanger')).toHaveAttribute('src', 'test-image-url');
  });

  it('calls showDetails with correct id on button click', () => {
    const mockShowDetails = jest.fn();
    const { getByTestId } = render(
      <ExchangerCard exchanger={mockExchanger} showDetails={mockShowDetails} />,
    );

    fireEvent.click(getByTestId('exchanger-button'));
    expect(mockShowDetails).toHaveBeenCalledTimes(1);
    expect(mockShowDetails).toHaveBeenCalledWith('test-id');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <ExchangerCard exchanger={mockExchanger} showDetails={() => {}} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct content', () => {
    const { getByText, getByAltText, getByTestId } = render(
      <ExchangerCard exchanger={mockExchanger} showDetails={() => {}} />,
    );

    expect(getByAltText('Test Exchanger')).toHaveAttribute('src', 'test-image-url');
    expect(getByText('Test Exchanger')).toBeInTheDocument();
    expect(getByText('1')).toBeInTheDocument(); // "Ranking: 1" was changed to "1"
    expect(getByTestId('exchanger-button')).toBeInTheDocument();
  });
});
