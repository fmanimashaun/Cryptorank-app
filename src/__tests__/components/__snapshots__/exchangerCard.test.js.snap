// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangerCard Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="card card"
  >
    <img
      alt="Test Exchanger"
      class="card-img image"
      src="test-image-url"
    />
    <div
      class="body card-body"
    >
      <div
        class="card-title h5"
      >
        Test Exchanger
      </div>
      <p
        class="text card-text"
      >
        Ranking: 1
      </p>
      <button
        class="btn btn btn-primary"
        type="button"
      >
        See Details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
